### DISCUSSION TITLE
Forcing dark mode on a container using responsive breakpoints doesn't work

### DISCUSSION THREAD

**Discussion by royliu2** [tailwindlabs/tailwindcss#12806] [Help]
Using:
- tailwind `3.4.1`
- Next `14.1.0`
- Next-themes `0.2.1`

Currently I can add `darkMode: "class"` to my tailwind.config.ts and use the `dark` class to force darkmode theme within a specific container. This is working great. However, if I want to use this class in combination with a responsive breakpoint, it no longer seems to work

```html
// This works for all breakpoints:
<div className='dark'>
  ...
</div>

// This does does not work for any breakpoints:
<div className='sm:dark'>
  ...
</div>
```

I would expect the latter example to force dark mode for the container for all breakpoints above `sm` but it does not. Has anyone run into similar issues or have idea how to resolve?

**Comment by wongjn** [comment]
You could consider:
- Using a [`MediaQueryList`](https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList) in JavaScript to dynamically add/remove the `dark` class as required.
- Using a [`variant` `darkMode` selector](https://github.com/tailwindlabs/tailwindcss/blob/7361468f77500105b0559e879e121f34306e8da2/src/corePlugins.js#L231-L262) like `'@media screen(sm) {.sm\\:dark &'` (as per [this Tailwind Play](https://play.tailwindcss.com/HNmTIRlOQ6?file=config), open the *Generated CSS* panel), though this means *only* `sm:dark` works – `dark` will no longer do anything.
- Create [your own variants with breakpoint prefixes](https://tailwindcss.com/docs/plugins#adding-variants) (as per [this Tailwind Play](https://play.tailwindcss.com/rZBtEZtULO?file=config), open the *Generated CSS* panel), though this may balloon the size of your CSS and you'll need to use a slightly different prefix since we cannot override the core `dark:` variant.
- Use the `sm:` variant on the "child" classes, like `sm:dark:text-white`.

**Reply by royliu2** [reply]
Thanks for the suggestions.

>Using a [MediaQueryList](https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList) in JavaScript to dynamically add/remove the dark class as required.

For our use-case we need to make sure this works properly with server components so I don't believe this approach will work.

>Using a [variant darkMode selector](https://github.com/tailwindlabs/tailwindcss/blob/7361468f77500105b0559e879e121f34306e8da2/src/corePlugins.js#L231-L262) like '@media screen(sm) {.sm\\:dark &' (as per [this Tailwind Play](https://play.tailwindcss.com/HNmTIRlOQ6?file=config), open the Generated CSS panel), though this means only sm:dark works – dark will no longer do anything.

We use `dark` elsewhere in our app so this is also a no-go.

>Use the sm: variant on the "child" classes, like sm:dark:text-white.

Doesn't this equate to - "starting from sm, use text-white _when in dark mode_"? we'd want something more like, "starting from sm, use the theme defined dark mode text color regardless of light or dark mode". 

I'll do some exploration of your custom variants suggestion to see if that works for us
